File: LDC.Atlas.Services.PreAccounting/Entities/Cost.cs
using System;

namespace LDC.Atlas.Services.PreAccounting.Entities
{
    public class Cost
    {
         public int CostId { get; set; }

         public int SectionId { get; set; }

         public string ContractLabel { get; set; }

         public string CostTypeCode { get; set; }

         public string SupplierCode { get; set; }

         public string Currency { get; set; }

         public string RateType { get; set; }

         public decimal Rate { get; set; }

         public string PricingMethod { get; set; }

         public int CostDirection { get; set; }

         public decimal OriginalEstimatePMTValue { get; set; }

         public decimal OriginalEstimateFullValue { get; set; }

         public decimal CalculatedFullValue { get; set; }

        public InvoiceStatus InvoiceStatus { get; set; }

        public string CreatedBy { get; set; }

        public DateTime CreationDate { get; set; }

         public string LastModifiedBy { get; set; }

         public DateTime LastModifiedDate { get; set; }

         public decimal AmountInvoiced { get; set; }

         public string Charter { get; set; }

         public decimal Quantity { get; set; }
    }
}

File: LDC.Atlas.WebApp/Client/app/shared/services/execution/dtos/cost.ts
import { InvoiceStatus } from '../../../enums/invoice-status.enum';

export class SearchCostForInvoicing {
	costType: string;
	supplierCode: string;
	charter: string;
	contractRef: string;

	constructor(costType: string, supplierCode: string, charter: string, contractRef: string) {
		this.costType = costType;
		this.supplierCode = supplierCode;
		this.charter = charter;
		this.contractRef = contractRef;
	}
}

export interface GetCostsRequest {
	costId: number;
	sectionId: number;
	contractLabel: string;
	costTypeCode: string;
	supplierCode: string;
	currency: string;
	rateType: number; // lumpSum / rate

	rate: number; // amount PMT
	pricingMethod: number;
	costDirection: number; // Pay / Rec
	quantity: number; // of the contract associated
	calculatedFullValue: number; // up to date full value of the cost

	charter: string; // the charter the contract is assigned to

	originalEstimatePMTValue: number; // original values that don't change once the cost/contract is approved
	originalEstimateFullValue: number; // original values that don't change once the cost/contract is approved

	amountInvoiced: number;
	amountToInvoice: number;

	invoiceStatus: InvoiceStatus;

	createdBy: string;
	creationDate: Date;
	lastModifiedBy: string;
	lastModifiedDate: Date;
}
